toolkit/primitives/domain/src/lib.rs

```diff
@@ -258,6 +256,20 @@ const MAINCHAIN_PRIVATE_KEY_LEN: usize = 32;
 #[byte_string(hex_serialize, hex_deserialize)]
 pub struct MainchainPrivateKey(pub [u8; MAINCHAIN_PRIVATE_KEY_LEN]);
 
+impl MainchainPrivateKey {
+	#[cfg(feature = "std")]
+	pub fn sign(&self, unsigned_message: &[u8]) -> Self {
+		Self(
+			ed25519::Pair::from_seed_slice(&self.0)
+				.expect("Mainchain key should be of correct length")
+				.sign(unsigned_message)
+				.0[0..64]
```

Comment:
I am not sure what 64 stands for, can we add some context here why signature is twice length?